pkg/plugins/manager.go

```diff
@@ -0,0 +1,172 @@
+//
+// Copyright 2025 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugins
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/rs/zerolog"
+)
+
+// Manager handles loading and managing plugins.
+type Manager struct {
+	plugins       map[string]*LoadedPlugin
+	pluginClients map[string]*plugin.Client
+	logger        zerolog.Logger
+}
+
+// LoadedPlugin represents a loaded plugin with its metadata.
+type LoadedPlugin struct {
+	Path     string
+	Plugin   Plugin
+	Metadata PluginMetadata
+}
+
+// NewManager creates a new plugin manager.
+func NewManager(logger zerolog.Logger) *Manager {
+	return &Manager{
+		plugins:       make(map[string]*LoadedPlugin),
+		pluginClients: make(map[string]*plugin.Client),
+		logger:        logger,
+	}
+}
+
+// LoadPlugins loads all plugins from the plugins directory.
+func (m *Manager) LoadPlugins(ctx context.Context) error {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("failed to get user home directory: %w", err)
+	}
+
+	pluginsDir := filepath.Join(homeDir, ".config", "chainloop", "plugins") // TODO: make this configurable
```

Comment:
btw, in the main CLI we use something like `	configPath := filepath.Join(xdg.ConfigHome, appName)`, you can do the same or extract that to a function